src/economy.cpp

```diff
@@ -2040,17 +2036,14 @@ CommandCost CmdBuyShareInCompany(DoCommandFlag flags, TileIndex tile, CompanyID
 
 	cost.AddCost(CalculateCompanyValue(c) >> 2);
 	if (flags & DC_EXEC) {
-		Owner *b = c->share_owners;
-
-		while (*b != COMPANY_SPECTATOR) b++; // share owners is guaranteed to contain at least one COMPANY_SPECTATOR
-		*b = _current_company;
-
-		for (int i = 0; c->share_owners[i] == _current_company;) {
-			if (++i == 4) {
-				c->bankrupt_value = 0;
-				DoAcquireCompany(c);
-				break;
-			}
+		auto unowned_share = std::find(c->share_owners.begin(), c->share_owners.end(), INVALID_OWNER);
+		assert(unowned_share != c->share_owners.end()); // share owners is guaranteed to contain at least one INVALID_OWNER, i.e. unowned share
+		*unowned_share = _current_company;
+
+		auto current_company_owns_share = [](auto share_owner) { return share_owner == _current_company; };
```

Comment:
I think _current_company should be captured in this lambda.

Reply:
`_current_company` is a global, it does not need a capture. No idea if a capture is optionally even allowed.

Reply:
Clang is quite resolute about it not being allowed. GCC "only" warns for it.
```
/home/rubidium/openttd/src/economy.cpp:2043:39: error: '_current_company' cannot be captured because it does not have automatic storage duration
                auto current_company_owns_share = [&_current_company](auto share_owner) { return share_owner == _current_company; };
                                                    ^
/home/rubidium/openttd/src/company_func.h:34:18: note: '_current_company' declared here
extern CompanyID _current_company;
                 ^
1 error generated.
```

Reply:
ok, thank you for checking :)